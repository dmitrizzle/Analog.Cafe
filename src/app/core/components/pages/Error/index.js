import React from "react"

import ArticleSection from "../Article/components/ArticleSection"
import ArticleWrapper from "../Article/components/ArticleWrapper"
import HeaderLarge from "../../vignettes/HeaderLarge"
import MetaTags from "../../vignettes/MetaTags"

export default props => {
  return (
    <ArticleWrapper>
      <MetaTags metaTitle={props.errorTitle} />
      <HeaderLarge
        pageTitle={props.errorTitle}
        pageSubtitle={props.errorSubtitle}
        title={props.errorDetails}
      />
      <ArticleSection>
        <p style={{ textAlign: "center" }}>{props.children}</p>
      </ArticleSection>
    </ArticleWrapper>
  )
}
