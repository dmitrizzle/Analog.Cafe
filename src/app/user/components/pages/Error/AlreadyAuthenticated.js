import React from "react"

import { withRouter } from "react-router"

import { CARD_ERRORS } from "../../../../core/constants/messages-"
import { ROUTE_URL_USER_LANDING } from "../../../constants/routes-session"
import { TEXT_ERRORS } from "../../../../constants"
import ErrorPage from "../../../../core/components/vignettes/ErrorPage"
import Link from "../../../../core/components/controls/Link"

class AlreadyAuthenticated extends React.PureComponent {
  componentWillMount = () => {
    this.props.history.replace({
      state: {
        status: "103" // already authenticated
      }
    })
  }
  componentWillUnmount = () => {
    this.props.history.replace({
      state: {
        status: "200"
      }
    })
  }
  render = () => {
    return (
      <ErrorPage
        errorTitle={CARD_ERRORS.ARTICLE.title}
        errorSubtitle={CARD_ERRORS.ARTICLE.subtitle}
        errorDetails={TEXT_ERRORS.CODE_103.error}
      >
        You are aloready signed in. Click{" "}
        <strong>
          <Link to={ROUTE_URL_USER_LANDING}>here</Link>
        </strong>{" "}
        to see your stuff.
      </ErrorPage>
    )
  }
}

export default withRouter(AlreadyAuthenticated)
