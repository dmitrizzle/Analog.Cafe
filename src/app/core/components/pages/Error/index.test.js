import "jest-styled-components"

import { shallow } from "enzyme"
import React from "react"

import Error from "./"

it("Render Error, matches snapshot", () => {
  const element = shallow(<Error />)
  expect(element).toMatchSnapshot()
})
