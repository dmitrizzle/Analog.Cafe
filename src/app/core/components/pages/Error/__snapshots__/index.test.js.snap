// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Error, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Unknown
          metaTitle={undefined}
/>,
        <Unknown
          pageSubtitle={undefined}
          pageTitle={undefined}
          title={undefined}
/>,
        <styled.section>
          <p
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          />
          <Connect(Footer) />
</styled.section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "metaTitle": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pageSubtitle": undefined,
          "pageTitle": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <p
              style={
                            Object {
                                          "textAlign": "center",
                                        }
              }
/>,
            <Connect(Footer) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Unknown
            metaTitle={undefined}
/>,
          <Unknown
            pageSubtitle={undefined}
            pageTitle={undefined}
            title={undefined}
/>,
          <styled.section>
            <p
                        style={
                                    Object {
                                                "textAlign": "center",
                                              }
                        }
            />
            <Connect(Footer) />
</styled.section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "metaTitle": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pageSubtitle": undefined,
            "pageTitle": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <p
                style={
                                Object {
                                                "textAlign": "center",
                                              }
                }
/>,
              <Connect(Footer) />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "style": Object {
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
